htlcswitch/interceptable_switch.go

```diff
@@ -119,10 +119,15 @@ type FwdResolution struct {
 	// FwdActionSettle.
 	Preimage lntypes.Preimage
 
-	// FailureMessage is the encrypted failure message that is to be passed
-	// back to the sender if action is FwdActionFail.
+	// FailureMessage is the failure message that is to be passed back to
+	// the sender if action is FwdActionFail.
 	FailureMessage []byte
 
+	// FailureMessageEncrypted indicates whether FailureMessage is already
+	// encrypted for the first hop and just needs to be intermediate-node
+	// obfuscated.
+	FailureMessageEncrypted bool
```

Comment:
should this rather be "is last hop" or similar?

Reply:
I'd rather keep it more low-level and only describe the contents of the failure message field.